pkg/scheduler/profile/profile.go

```diff
@@ -86,20 +85,21 @@ type cfgValidator struct {
 	queueSortArgs runtime.Object
 }
 
-func (v *cfgValidator) validate(cfg config.KubeSchedulerProfile) error {
-	if len(cfg.SchedulerName) == 0 {
+func (v *cfgValidator) validate(cfg config.KubeSchedulerProfile, f framework.Framework) error {
+	if len(f.ProfileName()) == 0 {
 		return errors.New("scheduler name is needed")
 	}
 	if cfg.Plugins == nil {
-		return fmt.Errorf("plugins required for profile with scheduler name %q", cfg.SchedulerName)
+		return fmt.Errorf("plugins required for profile with scheduler name %q", f.ProfileName())
 	}
-	if v.m[cfg.SchedulerName] != nil {
-		return fmt.Errorf("duplicate profile with scheduler name %q", cfg.SchedulerName)
+	if v.m[f.ProfileName()] != nil {
+		return fmt.Errorf("duplicate profile with scheduler name %q", f.ProfileName())
 	}
-	if len(cfg.Plugins.QueueSort.Enabled) != 1 {
-		return fmt.Errorf("one queue sort plugin required for profile with scheduler name %q", cfg.SchedulerName)
+	if len(f.ListPlugins().QueueSort.Enabled) != 1 {
```

Comment:
Better validate this in `NewFramework`

Reply:
Do you mean to add an additional check in `NewFramework` for the same thing?

Reply:
No, just remove this check here and do it in NewFramework.

Reply:
updated